flang/tools/tco/tco.cpp

```diff
@@ -116,7 +116,7 @@ static int compileFIR(const mlir::PassPipelineCLParser &passPipeline) {
   // run the pass manager
   if (mlir::succeeded(pm.run(*owningRef))) {
     // passes ran successfully, so keep the output
-    if (emitFir)
+    if (emitFir || passPipeline.hasAnyOccurrences())
       printModuleBody(*owningRef, out.os());
```

Comment:
I'm a little lost as to what this is doing now.

tco should be dumping LLVM IR output by default. The default output should be legal input to an LLVM tool like llc.

The `--emit-fir` flag was meant to simply parse the input and print it back out and exit. This was intended to test the FIR parser and pretty print only.

Reply:
This is also for testing, previously I added the option to run only the passes specified through command line. On Line 90 we check if any command line specifies any pass to run (pass pipeline only contains those) and if so it doesn't run the pass to dump llvm ir as we don't even generate llvm IR and thus we have to print the resulting module after running cli pass. This was needed for running the `Fir/affine-loop-fusion.fir` test through filecheck.

Reply:
Thank you for the explanation.